### ISSUE TITLE
 Segmentation fault

### ISSUE THREAD

**Issue by poetry236428** [fate1997/LineEvo#2]
We have a Segmentation fault that requires your assistance . We have tried different versions of pytorch (1.11.0,1.12.1+Cu102)
and Python 3.7, 3.8, 3.9, and 3.10, all of which are the same error. The dataset has been downloaded (104K bace.csv, 112K
bbbp.csv，36K delanev.csv，16K freesolv.csv，220K lipo.csv, which are in ./dataset/raw files) and the code is also your
original code. We hope you can provide us with some suggestions or solutions.

python train main.py -desc testing -model Add -dataset freesol
2023-08-30 15:58:22.344 -[INF]  ############testing############
2023-08-30 15:58:22 345 - INFO) - desc' 'testing, train': num epochs': 10000, r: 0.002, patience' 500,'save folder': "/results/'
task''regression’, 'metric: RMSE, 'device' 'cuda:0, og interval': 10), 'model: finput dim' 63, "hidden dim': 128, output dim': 1
num heads' 8. 'num lavers' 3'dropout': 02. redout' name' Add' 'dropout:0.1 num lavers' 1 "f pos: Truem. 'data'
('data name' 'freesolv', 'scaffold' False, 'seed' 42, batch size' 128, train ratio': 0.8
2023-08-30 15:58:22345-INFO1
-Dataset Loading---
data processing:   0%|                                                             | 0/642 [00:00<?, ?it/s]
Segmentation fault

**Comment by fate1997** [comment]
This may due to the missing folders in the `dataset` and `results` dirs. I have add some codes to the `train_main.py`, you may try again, and leaving me message if you still have problem. 

By the way, I noticed `python train main.py -desc testing -model Add -dataset freesol` is incorrect (maybe just a copy-paste error), and the correct command is `python train_main.py -desc testing -model Add -dataset freesolv`

**Comment by poetry236428** [comment]
I find parameter: testing is not find in configs/train.yml , parameter: dataset , I find  bace.csv  bbbp.csv  delaney.csv  freesolv.csv  lipo.csv in raw_files , and there are only python file in model file. But I can't find parameter: model named Add,can you tell me how to get the model?
[raw_files]$ du -sh *
104K    bace.csv
112K    bbbp.csv
36K     delaney.csv
16K     freesolv.csv
220K    lipo.csv
 [model]$ ls
GATv2_base.py  GNN_base.py  LineEvo_GATv2.py    LineEvo_schnet.py  __pycache__
GIN_base.py    __init__.py  LineEvo_readout.py  modules.py         SchNet_base.py

**Comment by fate1997** [comment]
1. `testing` is a given value to the `desc` argument in `train.yml`. (which is only for saving files' names, you can type any strings for this argument).
2. the dataset files' locations and names are correct.
3. `Add` is an optional value in the `model.readout.name` in `train.yml`, which means the normal `GATv2` or `GIN` model, you can also specify `LineEvo` to train a model proposed by this paper.

Sorry for the unclear usage information, I have added more information in the `README.md`. I have tested the codes on my machine and it works fine. If you still encounter problem, please attach the command and error message.
